digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LaneFiller.scala

```diff
@@ -0,0 +1,422 @@
+    val mValueAdjustments = (adjustedStartMeasure, adjustedEndMeasure) match {
+      case (None, None) => Nil
+      case (s, e)       => Seq(MValueAdjustment(lane.id, lane.linkId, s.getOrElse(lane.startMeasure), e.getOrElse(lane.endMeasure)))
+    }
+    val adjustedAsset = lane.copy(
+      startMeasure = adjustedStartMeasure.getOrElse(lane.startMeasure),
+      endMeasure = adjustedEndMeasure.getOrElse(lane.endMeasure))
+    (adjustedAsset, mValueAdjustments)
+  }
+
+
+  private def adjustLanesSideCodes(roadLink: RoadLink, lanes: Seq[PersistedLane], changeSet: ChangeSet): (Seq[PersistedLane], ChangeSet) = {
+
+    // auxiliary function to create PersistedLane object
+    def createPersistedLane(laneCode: Int, sideCode: Int, municipalityCode: Long, baseProperties: Seq[LaneProperty] ): PersistedLane = {
+      val lanePropertiesValues = LanePropertiesValues(baseProperties ++ Seq(LaneProperty("lane_code", Seq(LanePropertyValue(laneCode)))))
+
+      PersistedLane(0L, roadLink.linkId, sideCode, laneCode, municipalityCode,
+        0, roadLink.length, None, None, None, None, expired = false, roadLink.vvhTimeStamp, None,
+        lanePropertiesValues)
+    }
+
+
+    if(lanes.isEmpty )
+      return (lanes, changeSet)
+
+    val lanesToProcess = lanes.filter(_.linkId == roadLink.linkId)
+    val baseLane = lanesToProcess.sortBy(_.laneCode).head
+    val baseProps = baseLane.attributes.properties.filterNot(_.publicId == "lane_code")
+
+    roadLink.trafficDirection match {
+
+      case TrafficDirection.BothDirections =>
+
+        val mainLanes = ( lanesToProcess.exists(_.laneCode == 11),  lanesToProcess.exists(_.laneCode == 21) )
+
+        val toAdd = mainLanes match {
+          case (true, true) => Seq()
+          case (false, true) => Seq( createPersistedLane(11, SideCode.TowardsDigitizing.value, baseLane.municipalityCode, baseProps ) )
+
+          case (true, false) => Seq( createPersistedLane(21, SideCode.AgainstDigitizing.value, baseLane.municipalityCode, baseProps ) )
+          case (false, false) => Seq(
+                                createPersistedLane(11, SideCode.TowardsDigitizing.value, baseLane.municipalityCode, baseProps ),
+                                createPersistedLane(21, SideCode.AgainstDigitizing.value, baseLane.municipalityCode, baseProps )
+                              )
+          case _ => Seq()
+        }
+
+        (lanes ++ toAdd, changeSet.copy( generatedPersistedLanes = changeSet.generatedPersistedLanes ++ toAdd ))
+
+      case TrafficDirection.TowardsDigitizing =>
+
+        val toAdd = if ( !lanesToProcess.exists(_.laneCode == 11 ))
+                    Seq( createPersistedLane(11, SideCode.TowardsDigitizing.value, baseLane.municipalityCode, baseProps) )
+                  else
+                    Seq()
+
+
+        val toRemove = lanesToProcess.filter( lane => lane.laneCode >= 21 && lane.laneCode <= 29)
```

Comment:
Like was talked together with the team, let's implement some code values to avoid the propagation of these values like 21 or 29 directly on all code.